components/src/preact/mutationFilter/mutation-filter.stories.tsx

```diff
@@ -43,13 +43,77 @@ export const Default: StoryObj<MutationFilterProps> = {
     },
 };
 
+export const FiresFilterMultipleCommaSeparatedQueries: StoryObj<MutationFilterProps> = {
+    ...Default,
+    play: async ({ canvasElement, step }) => {
+        const { canvas, changedListenerMock } = await prepare(canvasElement, step);
+
+        await step('Enter a valid mutation', async () => {
+            await submitMutation(canvas, 'A123T');
+
+            await waitFor(() =>
+                expect(changedListenerMock).toHaveBeenCalledWith(
+                    expect.objectContaining({
+                        detail: {
+                            nucleotideMutations: ['A123T'],
+                            aminoAcidMutations: [],
+                            nucleotideInsertions: [],
+                            aminoAcidInsertions: [],
+                        },
+                    }),
+                ),
+            );
+        });
+
+        await step('Enter a comma separated list of valid and invalid mutations', async () => {
+            await pasteMutations(canvas, 'A123T, error_insX, A234T, ins_123:AA');
+
+            await waitFor(() =>
+                expect(changedListenerMock).toHaveBeenCalledWith(
+                    expect.objectContaining({
+                        detail: {
+                            nucleotideMutations: ['A123T', 'A234T'],
+                            aminoAcidMutations: [],
+                            nucleotideInsertions: ['ins_123:AA'],
+                            aminoAcidInsertions: [],
+                        },
+                    }),
+                ),
+            );
+            const firstMutationButtons = canvas.queryAllByText('A123T', { exact: false });
+            expect(firstMutationButtons).toHaveLength(1);
+            const secondValidMutationButtons = canvas.queryAllByText('A234T', { exact: false });
+            expect(secondValidMutationButtons).toHaveLength(1);
+            expect(inputField(canvas)).toHaveValue('error_insX');
```

Comment:
Those expects need to be awaited:
```suggestion
            await expect(firstMutationButtons).toHaveLength(1);
            const secondValidMutationButtons = canvas.queryAllByText('A234T', { exact: false });
            await expect(secondValidMutationButtons).toHaveLength(1);
            await expect(inputField(canvas)).toHaveValue('error_insX');
```